File: src/test/java/se/johannesdahlgren/aoc23/Day2Test.java
package se.johannesdahlgren.aoc23;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class Day2Test {

  @Test
  void part1Example() {
    Day2 day2 = new Day2("day2example.txt");
    int sum = day2.sumOfPossibleGameId(12, 13, 14);
    assertThat(sum).isEqualTo(8);
  }

  @Test
  void part1() {
    Day2 day2 = new Day2("day2.txt");
    int sum = day2.sumOfPossibleGameId(12, 13, 14);
    assertThat(sum).isEqualTo(2449);
  }

  @Test
  void part2Example() {
    Day2 day2 = new Day2("day2example.txt");
    int sum = day2.sumOfPowerOfMinNumberOfCubes();
    assertThat(sum).isEqualTo(2286);
  }

  @Test
  void part2() {
    Day2 day2 = new Day2("day2.txt");
    int sum = day2.sumOfPowerOfMinNumberOfCubes();
    assertThat(sum).isEqualTo(63981);
  }
}

File: src/test/kotlin/se/johannesdahlgren/aoc22/Day2Test.kt
package se.johannesdahlgren.aoc22

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day2Test {

    @Test
    fun part1Example() {
        val day2 = Day2("day2Example")
        assertThat(day2.getTotalScore()).isEqualTo(15)
    }

    @Test
    fun part1() {
        val day2 = Day2("day2")
        assertThat(day2.getTotalScore()).isEqualTo(12855)
    }

    @Test
    fun part2Example() {
        val day2 = Day2("day2Example")
        assertThat(day2.getTotalScoreFollowGuide()).isEqualTo(12)
    }

    @Test
    fun part2() {
        val day2 = Day2("day2")
        assertThat(day2.getTotalScoreFollowGuide()).isEqualTo(13726)
    }
}